endstate_correction/simulation/base.py

```diff
@@ -99,15 +99,7 @@ def _createSystem(self, **kwargs):
         return mm_system
 
     def start(self):
-        n_steps_per_sample = self.protocol.restart_interval
-
-        n_samples = int(
-            (
-                (self.protocol.runtime * unit.nanoseconds)
-                / (self.protocol.timestep * unit.femtoseconds)
-            )
-            / n_steps_per_sample
-        )
+        n_samples = int(self.protocol.runtime / self.protocol.timestep)
```

Comment:
I think it might be good to not hardcode the number of steps taken between each sample.

Reply:
that has not been resolved, I will get back to this tomorrow!